docs/develop/python/interceptors.mdx

```diff
@@ -0,0 +1,193 @@
+---
+id: interceptors
+title: Interceptors - Python SDK
+sidebar_label: Interceptors
+description: Learn how to implement Interceptors in the Temporal Python SDK to manage inbound and outbound SDK calls, enhance tracing, and add authorization to your Workflows and Activities.
+toc_max_heading_level: 2
+keywords:
+  - interceptors
+tags:
+  - Interceptors
+  - Python SDK
+  - Temporal SDKs
+---
+
+The behavior of the Python SDK can be customized in many useful ways by modifying inbound and outbound calls using Interceptors.
+This is similar to the use of "middleware" in web frameworks such as [Django](https://docs.djangoproject.com/en/5.2/topics/http/middleware/), [Starlette](https://www.starlette.io/middleware/), and [Flask](https://flask.palletsprojects.com/en/stable/lifecycle/#middleware).
+
+The methods you implement on your Interceptor classes can perform arbitrary side effects.
+Interceptors can also perform arbitrary modifications to data before it is received by the SDK's "real" implementation, e.g. for context propagation.
+
+There are five categories of inbound and outbound calls that you can modify in this way:
+
+#### [Outbound Client calls](https://python.temporal.io/temporalio.client.OutboundInterceptor.html)
+
+- `start_workflow()`
+- `signal_workflow()`
+- `list_workflows()`
+- `update_schedule()`
+
+This is not an exhaustive list; refer to the [Python SDK methods](https://python.temporal.io/temporalio.client.OutboundInterceptor.html) for more.
+
+#### [Inbound Workflow calls](https://python.temporal.io/temporalio.worker.WorkflowInboundInterceptor.html)
+
+- `execute_workflow()`
+  - Note that `execute_workflow()` is considered an Inbound Workflow call unlike `start_workflow()` because it blocks and waits for a result to be returned.
```

Comment:
`execute_workflow` here is not really related to the `start_workflow` vs `execute_workflow` distinction in the client API. `execute_workflow` is just the name given to the task that a workflow has to do when it is told to execute a workflow.

It's like "handle a request to execute user workflow code".

Mentioning the client methods will confuse readers. Could maybe be something like


```suggestion
- `execute_workflow()` (i.e. handle a workflow task that is starting a new Workflow Execution)
```